### PR TITLE
Reintroduce game-engines/renpy

### PR DESCRIPTION
**Pull request by lanodan**
RenPy 8 got released and is based on Python3 with compatibility for previous games.

This was in my overlay and been using it for some months.

- dev-python/pygame_sdl2: add 8.0.2.22081402
- profiles: Only mask old dev-python/pygame_sdl2
- games-engines/renpy: new package, add 8.0.3

### CODE REVIEW

File: dev-python/pygame_sdl2/pygame_sdl2-8.0.2.22081402.ebuild [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
Comment (thesamesam) [inline dev-python/pygame_sdl2/pygame_sdl2-8.0.2.22081402.ebuild:4]: 8, please

File: dev-python/pygame_sdl2/pygame_sdl2-8.0.2.22081402.ebuild [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+PYTHON_COMPAT=( python3_{8,9,10} )
```
Comment (thesamesam) [inline dev-python/pygame_sdl2/pygame_sdl2-8.0.2.22081402.ebuild:6]: 8 is gone and a noop. Also, no 3.11?

File: dev-python/pygame_sdl2/pygame_sdl2-8.0.2.22081402.ebuild [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+inherit distutils-r1
```
Comment (thesamesam) [inline dev-python/pygame_sdl2/pygame_sdl2-8.0.2.22081402.ebuild:7]: No PEP517?

File: dev-python/pygame_sdl2/pygame_sdl2-8.0.2.22081402.ebuild [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+# setuptools 67.2.0 (https://bugs.gentoo.org/892347) and includes pregenerated
+# files.
+SRC_URI="https://github.com/renpy/pygame_sdl2/archive/refs/tags/renpy-${PV}.tar.gz -> ${P}.tar.gz"
+S="${WORKDIR}/${PN}-renpy-${PV}/"
+
+# TODO: Debundle DejaVu
+LICENSE="LGPL-2.1 ZLIB BitstreamVera"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
```
Comment (thesamesam) [inline dev-python/pygame_sdl2/pygame_sdl2-8.0.2.22081402.ebuild:22]: Drop

File: games-engines/renpy/files/renpy-6.99.12.4-compat-infinite-loop.patch [hunk 5]
```diff
@@ -0,0 +1,42 @@
+Adapted from git HEAD:
```
Comment (thesamesam) [inline games-engines/renpy/files/renpy-6.99.12.4-compat-infinite-loop.patch:1]: Link to the source please.

File: games-engines/renpy/files/renpy-6.99.12.4-compat-style.patch [hunk 6]
```diff
@@ -0,0 +1,43 @@
+commit 1d350d993d2b51df7880e92b48ed103e059ec385
```
Comment (thesamesam) [inline games-engines/renpy/files/renpy-6.99.12.4-compat-style.patch:1]: Link to the source please.

File: games-engines/renpy/files/renpy-8.0.3-use-system-fribidi.patch [hunk 7]
```diff
@@ -0,0 +1,33 @@
+--- renpy-7.3.5-source.orig/module/setup.py	2019-08-09 05:11:58.000000000 +0300
```
Comment (thesamesam) [inline games-engines/renpy/files/renpy-8.0.3-use-system-fribidi.patch:1]: scrub-patch (iwdevtools)

File: games-engines/renpy/files/renpy.1 [hunk 8]
```diff
@@ -0,0 +1,163 @@
+.\" Copyright © 2015 Andrew Savchenko
```
Comment (thesamesam) [inline games-engines/renpy/files/renpy.1:1]: This should really be sent upstream.

File: games-engines/renpy/renpy-8.0.3.ebuild [hunk 9]
```diff
@@ -0,0 +1,111 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
Comment (thesamesam) [inline games-engines/renpy/renpy-8.0.3.ebuild:4]: - EAPI 8
- Drop Python 3.8 (it does nothing now)
- Add Python 3.11? See gentoo-dev ML too.

**Review by thesamesam — CHANGES_REQUESTED** [review]

**Comment by bircoph** [comment]
Hi!

I bumped pygame_sdl2 to 8.0.3. Could you please rebase and check that renpy-8.0.3 builds/works fine with it? I had to change build tag from "-for-renpy-8.0.3" to "+renpy8.0.3" to conform to PEP 440, see Gentoo bug 892347.

**Comment by bircoph** [comment]
You can also submit patches/updates through Gentoo bugzilla or e-mail to me if you don't like Github as I can see from your profile. Frankly, I missed this update as I rarely use Github for Gentoo.

Would you like to proxy-maint renpy in Gentoo? I don't have enough time to update and test it thoroughly these days, but I can try to proxy your contributions.

**Comment by lanodan** [comment]
>I bumped pygame_sdl2 to 8.0.3. Could you please rebase and check that renpy-8.0.3 builds/works fine with it? I had to change build tag from "-for-renpy-8.0.3" to "+renpy8.0.3" to conform to PEP 440, see Gentoo bug 892347.

Sadly setuptools also broke the way headers are installed in the process…

``/usr/include/python3.10/pygame_sdl2/pygame_sdl2.h`` (seems correct) vs. ``/usr/lib/python3.10/site-packages/pygame_sdl2-2.1.0.data/headers/pygame_sdl2.h``

Renpy uses ``#include <pygame_sdl2/pygame_sdl2.h>``

> You can also submit patches/updates through Gentoo bugzilla or e-mail to me if you don't like Github as I can see from your profile. Frankly, I missed this update as I rarely use Github for Gentoo.
> Would you like to proxy-maint renpy in Gentoo? I don't have enough time to update and test it thoroughly these days, but I can try to proxy your contributions.

I guess I could CC you on patches to gentoo-proxy-maint@ for next versions.

**Comment by thesamesam** [comment]
Could you rebase this to drop the pygame_sdl2 bit? That's fixed upstream now thanks to Eli.

Also, please update to latest renpy version. Hopefully the future dep is gone then..

**Comment by lanodan** [comment]
[2023-08-07 04:45:48+0000] Sam James:
>Could you rebase this to drop the pygame_sdl2 bit? That's fixed upstream now thanks to Eli.

Nice, shouldn't pygame_sdl2 get a revbump which adds DEPEND=">=dev-python/wheel-0.41.0"?

> Hopefully the future dep is gone then..

I doubt that's going to happen, AFAIK renpy8 uses it to be compatible with prior games written with python2.

**Pull request closed without merging**